app/models/booking.rb

```diff
@@ -0,0 +1,21 @@
+class Booking < ApplicationRecord
+  belongs_to :user
+  belongs_to :listing
+
+  STATUS = ["Accepted", "Pending", "Cancelled"]
+
+  validates :start_date, presence: true
+  validates :end_date, presence: true
+  validate :end_date_is_after_start_date
```

Comment:
I am using "validate" here because `end_date_is_after_start_date` is a custom method. "validates" works for default validators only, as explained [here](https://stackoverflow.com/questions/18140898/whats-the-difference-between-validate-and-validates) and as seen [here](https://devdocs.io/rails~6.1/activemodel/validations#method-i-validate).